src/cargo/core/features.rs

```diff
@@ -893,21 +924,17 @@ fn parse_gitoxide(
     let mut out = GitoxideFeatures::default();
     let GitoxideFeatures {
         fetch,
-        shallow_index,
         checkout,
-        shallow_deps,
         internal_use_git2,
     } = &mut out;
 
     for e in it {
         match e.as_ref() {
             "fetch" => *fetch = true,
-            "shallow-index" => *shallow_index = true,
-            "shallow-deps" => *shallow_deps = true,
```

Comment:
I'd like to hear @Byron's opinion of this :)

Reply:
Thanks for reeling me in! I like the unification of nightly feature toggles where possible, and I can totally get behind having a unified `-Zgit-shallow` for convenience.

However, since it's nightly, I still think that it would be worthwhile to have a toggle for `shallow-deps` and `shallow-index` separately just in case shallow-deps clone fail: it's nightly, after all, and dependencies could involve cloning submodules shallowly too and who knows what can go wrong when complexity is increased like that, our coverage on that is certainly limited, and the git2 implementation is quite new as well.